dodo.py

```diff
@@ -16,42 +16,92 @@
+            INPUT_FILES = os.path.join(LOCALE_PATH, "*.properties")
+            OUTPUT_FILE = os.path.join(LOCALE_PATH, "LC_MESSAGES", "combined.properties")
+
+            with open(OUTPUT_FILE, "w+", encoding="utf-8") as outfile:
+                for file in glob.glob(INPUT_FILES):
+                    print(file)
+                    with open(file, encoding="utf-8") as infile:
+                        outfile.write(infile.read())
+                        outfile.write("\n")
+            print("Combined strings for {}".format(current_locale))
+            sort_and_uniquify_file(OUTPUT_FILE)
+
+
+    
+    """
+    Delete strings that aren't used in the code, to keep size down and not waste time fixing unused strings
+    Input: combined.properties.sorted
+    Output: filtered.properties
+    """
+    def filter():
+        REF_FILE = os.path.join("tabcmd", "locales", "codestrings.properties")
+        for current_locale in LOCALES:
+            LOCALE_PATH = os.path.join("tabcmd", "locales", current_locale)
+            IN_FILE = os.path.join(LOCALE_PATH, "LC_MESSAGES", "combined.properties")
+            OUT_FILE = os.path.join(LOCALE_PATH, "LC_MESSAGES", "filtered.properties")
+            
+            with open(REF_FILE, "r", encoding="utf-8") as ref:
+                required = ref.read()
+                with open(IN_FILE, "r", encoding="utf-8") as infile, open(OUT_FILE, "w+", encoding="utf-8") as outfile:
+                    for line in infile.readlines():
+                        key=line.split("=")[0]
+                        if key in required:
+                            outfile.writelines(line)
+                            
+                            
+            print("Filtered strings for {}".format(current_locale))
+    
+    """
+    For all languages: Read filtered.properties files with unicode like "Schlie\u00dfen", save it back as "Schließen"
     requires: pip install ftfy
     help: https://ftfy.readthedocs.io/
-    Run when we copy in updated properties files. NOT IDEMPOTENT: only run on the initial files
+    
+    Inputs: locales/*_[locale]/LC_MESSAGES/filtered.properties files
+    Output: (generated clean each run) locales/[locale]/LC_MESSAGES/transcoded.properties file
     """
 
-    def process_locales():
+    def encode():
         for current_locale in LOCALES:
-            if current_locale in ["en", "ga"]:  # greek is our pseudo-loc
-                continue
-            elif current_locale in ["ja", "ko", "zh"]:
-                encoding = "utf-8"
-            else:
-                encoding = "cp1252"
-
-            for file in glob.glob("tabcmd/locales/" + current_locale + "/*.properties"):
-                basename = os.path.basename(file).split(".")[0]
-                print("transcoding", basename)
-                with open(file, encoding=encoding) as infile:
+            # I'm not sure why we were varying the locale before, but this seems to work fine
+            encoding = "utf-8"
```

Comment:
Maybe get rid of the `encoding` variable entirely. All the previous .properties files are currently using utf-8; maybe those 3 locales were previously using `Windows-1252` at some point.